clang/lib/Sema/SemaChecking.cpp

```diff
@@ -5967,8 +5967,10 @@ void Sema::checkCall(NamedDecl *FDecl, const FunctionProtoType *Proto,
     CheckSYCLKernelCall(FD, Range, Args);
 
   // Diagnose variadic calls in SYCL.
-  if (FD && FD ->isVariadic() && getLangOpts().SYCLIsDevice &&
-      !isUnevaluatedContext() && !isKnownGoodSYCLDecl(FD))
+  if (FD && FD->isVariadic() &&
+      FD->getBuiltinID() != Builtin::BI__builtin_assume_aligned &&
+      getLangOpts().SYCLIsDevice && !isUnevaluatedContext() &&
+      !isKnownGoodSYCLDecl(FD))
```

Comment:
I would like this builtin check moved inside the routine isKnownGoodSYCLDecl so the exceptions remain in one place.